Components/Helper/Currency.php

```diff
@@ -0,0 +1,178 @@
+ *
+ * 3. Neither the name of the copyright holder nor the names of its contributors
+ * may be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * @author Nosto Solutions Ltd <shopware@nosto.com>
+ * @copyright Copyright (c) 2018 Nosto Solutions Ltd (http://www.nosto.com)
+ * @license http://opensource.org/licenses/BSD-3-Clause BSD 3-Clause
+ */
+
+use Shopware_Plugins_Frontend_NostoTagging_Bootstrap as Bootstrap;
+use Shopware\Models\Shop\Shop;
+use Shopware\Models\Shop\Currency;
+use Nosto\Object\Format;
+
+/**
+ * Helper class for Currency
+ *
+ * @package Shopware
+ * @subpackage Plugins_Frontend
+ */
+class Shopware_Plugins_Frontend_NostoTagging_Components_Helper_Currency
+{
+    const CURRENCY_SYMBOL_LEFT = 32;
+    const CURRENCY_SYMBOL_RIGHT = 16;
+    const CURRENCY_SYMBOL_DEFAULT = 0;
+    const CURRENCY_DECIMAL_CHAR = ',';
+    const CURRENCY_GROUPING_CHAR = '.';
+    const CURRENCY_DECIMAL_PRECISION = 2;
+
+    /**
+     * Get currencies for the given shop.
+     * If it's a sub shop, returns the parent's currencies
+     *
+     * @param Shop $shop
+     * @return \Doctrine\Common\Collections\ArrayCollection|Currency[]
+     */
+    public static function getCurrencies(Shop $shop)
+    {
+        $currencies = $shop->getCurrencies();
+        if ($currencies->isEmpty()) {
+            // If it's a sub-shop, currencies are inherited from the main shop
+            $currencies = $shop->getMain()->getCurrencies();
+            if ($currencies->isEmpty()) {
+                /** @noinspection PhpUndefinedMethodInspection */
+                Shopware()->Plugins()->Frontend()->NostoTagging()->getLogger()->warning(
+                    sprintf(
+                        'Shop %s and his main shop have no currencies',
+                        $shop->getName()
+                    )
+                );
+            }
+        }
+        return $currencies;
+    }
+
+    /**
+     * @param Shop $shop
+     * @return string
+     */
+    public static function getCurrencyCode(Shop $shop)
+    {
+        if (self::isMultiCurrencyEnabled($shop)) {
```

Comment:
Why these nested lookups? 🤔

Reply:
In the Product Model we should only set the `variationId` if the Multi-Currency is enabled.
The `getCurrencyCode` returns the default for the store, or the main used by the shop in the front-end context